backoffice/utils.py

```diff
@@ -37,10 +37,18 @@ def get_course_key(course_key_string):
     return CourseKey.from_string(course_key_string)
 
 def group_required(*group_names):
-    """Requires user membership in at least one of the groups passed in."""
+    """
+    Requires user membership in at least one of the groups passed in.
+    The request user may be a staff user impersonating another user so we must
+    also check this case and allow access.
+    """
     def in_groups(user):
         if user.is_authenticated():
-            if bool(user.groups.filter(name__in=group_names)) or user.is_superuser:
+            if (
+                    user.is_superuser or
+                    bool(user.groups.filter(name__in=group_names)) or
+                    user.original_user.is_superuser or
+                    bool(user.original_user.groups.filter(name__in=group_names))):
```

Comment:
If I understand well, during impersonation, you keep original user permissions? In this case, it's not a real impersonation and could bias the impersonation process, no?

Reply:
True but this decorator is used in the backoffice so I keep the staff rights in the backoffice while he is impersonating the user on the LMS. So that's great and a full impersonation!

I should add this as a comment to make it clear.

Reply:
Ok, you are right: preventing back-office usage during impersonation would have led to negative feedback from staff users :stuck_out_tongue:

Reply:
although that's what they have now!

Reply:
I think you should add a permanent reminder of  impersonation in the backoffice.
`<div class="alert">warning: you are impersonating user <user></div>`
Probably here: https://github.com/openfun/fun-apps/blob/fun/release-4.19/backoffice/templates/backoffice/base.html#L56